products/batch_exports/backend/tests/temporal/destinations/s3/test_activity_with_gcs_bucket.py

```diff
@@ -0,0 +1,305 @@
+import os
+import uuid
+
+import pytest
+
+import aioboto3
+from temporalio.testing._activity import ActivityEnvironment
+
+from posthog.batch_exports.service import BatchExportModel, BatchExportSchema
+from posthog.temporal.tests.utils.events import generate_test_events_in_clickhouse
+
+from products.batch_exports.backend.temporal.destinations.s3_batch_export import (
+    COMPRESSION_EXTENSIONS,
+    FILE_FORMAT_EXTENSIONS,
+    SUPPORTED_COMPRESSIONS,
+    S3InsertInputs,
+    s3_default_fields,
+)
+from products.batch_exports.backend.tests.temporal.destinations.s3.utils import (
+    TEST_S3_MODELS,
+    _run_activity,
+    assert_clickhouse_records_in_s3,
+    has_valid_gcs_credentials,
+)
+from products.batch_exports.backend.tests.temporal.utils.s3 import (
+    assert_files_in_s3,
+    delete_all_from_s3,
+    read_json_file_from_s3,
+)
+
+pytestmark = [
+    pytest.mark.asyncio,
+    pytest.mark.django_db,
+    pytest.mark.skipif(
+        not has_valid_gcs_credentials(),
+        reason="GCS credentials (AWS keys) not set in environment or missing GCS_TEST_BUCKET variable",
+    ),
+]
+
+
+@pytest.fixture
+def bucket_name():
+    return os.getenv("GCS_TEST_BUCKET")
+
+
+@pytest.fixture
+async def gcs_client(bucket_name, s3_key_prefix):
+    """Manage an S3 client to interact with a GCS bucket."""
+    async with aioboto3.Session().client("s3", endpoint_url="https://storage.googleapis.com") as s3_client:
+        yield s3_client
+
+        await delete_all_from_s3(s3_client, bucket_name, key_prefix=s3_key_prefix)
+
+
+@pytest.mark.parametrize("compression", COMPRESSION_EXTENSIONS.keys(), indirect=True)
+@pytest.mark.parametrize("model", TEST_S3_MODELS)
+@pytest.mark.parametrize("file_format", FILE_FORMAT_EXTENSIONS.keys())
+async def test_insert_into_s3_activity_puts_data_into_gcs(
```

Comment:
perhaps slightly overkill to add all these tests for GCS given that we just want to test the manifest is uploaded correctly. I mean, it doesn't hurt to have full test coverage, it will just make the tests slower to run. I suppose you could argue you won't be running both GCS and S3 tests at the same time though